### PR TITLE
fix(x/intertx): fix ICA integration

### PR DESCRIPTION
**Pull request by aleem1314**
## Description

Found few issues, while testing ICA.

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/main/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/main/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed 
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: x/intertx/types/v1/msg_submit_tx.go [hunk 1]
```diff
@@ -63,3 +69,25 @@ func NewMsgSubmitTx(owner string, connectionID string, msg sdk.Msg) *MsgSubmitTx
...
+	any, err := codectypes.NewAnyWithValue(msg)
+	if err != nil {
+		return nil, err
+	}
+
+	return any, nil
+}
+
+// UnpackInterfaces implements codectypes.UnpackInterfacesMessage
+func (msg MsgSubmitTx) UnpackInterfaces(unpacker codectypes.AnyUnpacker) error {
```
Comment (aleem1314) [inline x/intertx/types/v1/msg_submit_tx.go:89]: Without this method, `submit-tx` returning error `cannot unpack Any into type ....`.

File: x/intertx/types/v1/msg_register_account_test.go [hunk 2]
```diff
@@ -23,6 +25,8 @@ func TestMsgRegisterAccount(t *testing.T) {
 
 func (s *registerAccountSuite) Before(t gocuke.TestingT) {
 	s.t = t
+	cfg := sdk.GetConfig()
```
Comment (aleem1314) [inline x/intertx/types/v1/msg_register_account_test.go:28]: This is strange. I am getting the `expected cosmos got regen` error.

File: x/intertx/types/v1/msg_submit_tx.go [hunk 3]
```diff
@@ -63,3 +69,25 @@ func NewMsgSubmitTx(owner string, connectionID string, msg sdk.Msg) *MsgSubmitTx
...
+	if !ok {
+		return nil, fmt.Errorf("can't proto marshal %T", sdkMsg)
+	}
+
+	any, err := codectypes.NewAnyWithValue(msg)
+	if err != nil {
+		return nil, err
+	}
+
+	return any, nil
```
Comment (technicallyty) [inline x/intertx/types/v1/msg_submit_tx.go:85]:
```suggestion
	return codectypes.NewAnyWithValue(msg)
```

**Review by technicallyty — APPROVED** [review]
LGTM, 2 small nits

**Comment by aleem1314** [comment]
@ryanchristo sample repo to test ICA between Regen and Akash network.
https://github.com/aleem1314/ica-testing

File: x/intertx/client/tx.go [hunk 4]
```diff
@@ -45,10 +45,13 @@ func GetTxCmd() *cobra.Command {
...
-		Use:     "register --connection-id <connection-id> --version <version>",
-		Short:   "register an interchain account",
-		Long:    "register an interchain account for the chain corresponding to the connection-id.",
-		Example: "regen tx intertx register --connection-id channel-10 --version v5",
+		Use:   "register --connection-id <connection-id> --version <version>",
+		Short: "register an interchain account",
+		Long:  "register an interchain account for the chain corresponding to the connection-id.",
+		Example: `
+		regen tx intertx register --connection-id channel-10
+		regen tx intertx register --connection-id channel-10 --version '{"version":"ics27-1","tx_type":"sdk_multi_msg","encoding":"proto3","host_connection_id":"connection-0","controller_connection_id":"connection-0","address":"regen14zs2x38lmkw4eqvl3lpml5l8crzaxn6mpvh79z"'
```
Comment (ryanchristo) [inline x/intertx/client/tx.go:53]:
```suggestion
		regen tx intertx register --connection-id channel-10 --version '{"version":"ics27-1","tx_type":"sdk_multi_msg","encoding":"proto3","host_connection_id":"connection-0","controller_connection_id":"connection-0","address":"regen14zs2x38lmkw4eqvl3lpml5l8crzaxn6mpvh79z"}'
```

File: x/intertx/types/v1/init_test.go [hunk 5]
```diff
@@ -0,0 +1,8 @@
+package v1
+
+import sdk "github.com/cosmos/cosmos-sdk/types"
+
+func init() {
+	cfg := sdk.GetConfig()
+	cfg.SetBech32PrefixForAccount("regen", "regenpub")
+}
```
Comment (ryanchristo) [inline x/intertx/types/v1/init_test.go:8]: I think we can remove this file. We have a `config.go` file in the root of `x/intertx` that should apply.

**Review by ryanchristo — APPROVED** [review]
tACK using https://github.com/aleem1314/ica-testing :rocket:

Just a couple small nits. Great work and thanks for digging into this!

File: x/intertx/types/v1/msg_submit_tx.go (hunk 1 above)
Comment (ryanchristo) [reply x/intertx/types/v1/msg_submit_tx.go:84]: I think this was an issue with the json example in testing. I was able to successfully test with the fix I submitted in your testing repo. Do we still need this?

File: x/intertx/types/v1/init_test.go (hunk 5 above)
Comment (aleem1314) [reply x/intertx/types/v1/init_test.go:8]: Removing this file causing V1 package tests fails with error `expected cosmos got regen` error.

**Pull request merged**